controllers/storagecluster/generate.go

```diff
@@ -45,10 +45,26 @@ func generateNameForCephBlockPool(initData *ocsv1.StorageCluster) string {
 }
 
 func generateDeviceClass(initData *ocsv1.StorageCluster) string {
-	if initData.Spec.ManagedResources.CephNonResilientPools.Enable {
-		return "replicated"
+	if len(foundDeviceClasses) == 0 {
```

Comment:
I don't think this condition will help us much, eventually, it will go in the next else if condition in the next reconcile once the status is set.